### PR TITLE
Improve props III in React

### PR DESCRIPTION
**Pull request by kapnobatai137**
componentwillreceiveprops-not-triggered-after-mounting:
- deprecated method, insight was archived

1. clone elements in React
- updated syntax to the latest version
- improved formatting of code blocks
- added example of passing props to children

2. shallow compare in react
- updated syntax to the latest version
- added example of creating a pure component
- explained what a pure component does, what a shallow compare is
- added rules/implications of using pure components
- alternative for functional components by using React.memo
- updated PQ and RQ to represent the new content

3. shortcut for transferring props
- updated syntax to the latest version

4. use proptypes on stateless components
- updated syntax to the latest version

### CODE REVIEW

**Comment by enki-hq** [comment]
Rule insight:code-width failed for files: 
  - javascript/react/props-iii/shallow-compare-in-react.md

**Comment by enki-hq** [comment]
### Insight previews:
- [ ] [enkidevs/curriculum/.archived/javascript/react/props-iii/componentwillreceiveprops-not-triggered-after-mounting.md](https://app.enkipro.com/#/public/preview-insight/enkidevs/curriculum/.archived/javascript/react/props-iii/componentwillreceiveprops-not-triggered-after-mounting.md?ref=748bc48129ed6defc33434612ca33e059eb3b85f)
- [ ] [enkidevs/curriculum/javascript/react/props-iii/clone-elements-in-react.md](https://app.enkipro.com/#/public/preview-insight/enkidevs/curriculum/javascript/react/props-iii/clone-elements-in-react.md?ref=748bc48129ed6defc33434612ca33e059eb3b85f)
- [ ] [enkidevs/curriculum/javascript/react/props-iii/shallow-compare-in-react.md](https://app.enkipro.com/#/public/preview-insight/enkidevs/curriculum/javascript/react/props-iii/shallow-compare-in-react.md?ref=748bc48129ed6defc33434612ca33e059eb3b85f)
- [ ] [enkidevs/curriculum/javascript/react/props-iii/shortcut-for-transferring-props.md](https://app.enkipro.com/#/public/preview-insight/enkidevs/curriculum/javascript/react/props-iii/shortcut-for-transferring-props.md?ref=748bc48129ed6defc33434612ca33e059eb3b85f)
- [ ] [enkidevs/curriculum/javascript/react/props-iii/use-proptypes-on-stateless-components.md](https://app.enkipro.com/#/public/preview-insight/enkidevs/curriculum/javascript/react/props-iii/use-proptypes-on-stateless-components.md?ref=748bc48129ed6defc33434612ca33e059eb3b85f)

File: javascript/react/props-iii/clone-elements-in-react.md [hunk 1]
```diff
@@ -26,42 +26,93 @@ aspects:
...
-  [children ...]
+```jsx
+React.cloneElement(
+  element,
+  [props],
+  [...children]
 );
 ```
 
 A new `ReactElement` will be returned using the provided `element` as the starting point. The old `props` will be merged in shallowly with the new specified ones. Furthermore, new `children` will replace old ones.
```
Comment (nem035) [inline javascript/react/props-iii/clone-elements-in-react.md]:
```suggestion
A new element will be returned using the provided element as the starting point. The old `props` will be merged in shallowly with the new specified ones. Furthermore, new `children` will replace old ones.
```

Comment (nem035) [reply javascript/react/props-iii/clone-elements-in-react.md]: We can add 2 footnotes:

1) Explaining the element similarly to [here](https://reactjs.org/docs/glossary.html#elements)
2) Explaining "shallow merge" with something along the lines of:

> A merge of two objects in JavaScript produces a new object by combining their properties. Properties that only exists in one of the two objects are just copied into the result. If a property exists in both, the value of that property from the second object will overwrite the value of the first. A"shallow" merge means that we only compare/merge top-level properties. Merging all levels of properties between two objects is called a *deep merge*.

Comment (kapnobatai137) [reply javascript/react/props-iii/clone-elements-in-react.md]: Added the following footnotes:

[1: element]
The React element represents the building block of any React application. Although one might confuse elements with components, they are not the same. An element is used to describe what you want to see on the screen. Here's an example:
```jsx
const element = <h1>Enki is cool!</h1>;
```

[2: shallow merge]
A merge of two objects in JavaScript produces a new object by combining their properties. Properties that only exists in one of the two objects are just copied into the result. If a property exists in both, the value of that property from the second object will overwrite the value of the first. A"shallow" merge means that we only compare/merge top-level properties. Merging all levels of properties between two objects is called a *deep merge*.

**Pull request closed without merging**